libsplinter/src/circuit/template/rules/create_services.rs

```diff
@@ -50,3 +128,182 @@ impl From<v1::ServiceArgument> for ServiceArgument {
         }
     }
 }
+
+fn get_next_service_id(current_id: &str) -> Result<String, CircuitTemplateError> {
+    let alphanumeric = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz";
+    let mut next_id = current_id.to_string();
+    for (char_index, char) in current_id.char_indices().rev() {
+        let index = alphanumeric.find(char).ok_or_else(|| {
+            CircuitTemplateError::new(
+                "The field first_service must contain only valid base62 characters",
+            )
+        })?;
+        match alphanumeric.get(index + 1..index + 2) {
+            Some(sub_str) => {
+                let next_id_sub_string = next_id.get(char_index + 1..).unwrap_or_default();
+                let new_sub_string = format!("{}{}", sub_str, next_id_sub_string);
+                next_id.replace_range(char_index.., &new_sub_string);
+                return Ok(next_id);
+            }
+            None => {
+                let next_id_sub_string = next_id.get(char_index + 1..).unwrap_or_default();
+                let new_sub_string =
+                    format!("{}{}", alphanumeric[0..1].to_string(), next_id_sub_string);
+                next_id.replace_range(char_index.., &new_sub_string);
+            }
+        }
+    }
+
+    Err(CircuitTemplateError::new(
+        "Exceed number of services that can be built",
+    ))
+}
```

Comment:
This could probably be simplified by just incrementing the chars (https://users.rust-lang.org/t/char-how-to-get-next-char-in-rust/17720), checking if the resulting char is alphanumeric (https://doc.rust-lang.org/std/primitive.char.html#method.is_ascii_alphanumeric), and overflowing to the next char if necessary.

Reply:
Looks good!